src/cli/build/assets.ts

```diff
@@ -0,0 +1,79 @@
+import path from 'path';
+
+import type { Color } from 'chalk';
+import chalk from 'chalk';
+import fs from 'fs-extra';
+
+import { copyFile } from '../../utils/copy';
+import { buildPatternToFilepathMap, crawlDirectory } from '../../utils/dir';
+import { type Logger, createLogger, log } from '../../utils/logging';
+import {
+  getConsumerManifest,
+  getPropFromConsumerManifest,
+} from '../../utils/manifest';
+
+export const copyAssets = async (
+  destinationDir: string,
+  logger: Logger = log,
+) => {
+  const manifest = await getConsumerManifest();
+  const assets = await getPropFromConsumerManifest<string, string[]>('assets');
+
+  if (!manifest || !assets) {
+    return;
+  }
+
+  const resolvedSrcDir = path.resolve(path.dirname(manifest.path), 'src');
+  const resolvedDestinationDir = path.resolve(
+    path.dirname(manifest.path),
+    destinationDir,
+  );
+  const allFiles = await crawlDirectory(resolvedSrcDir);
+  const filesByPattern = buildPatternToFilepathMap(assets, allFiles, {
+    cwd: resolvedSrcDir,
+    dot: true,
+  });
+
+  for (const filenames of Object.values(filesByPattern)) {
```

Comment:
Would a await promise.all map be better here?

Reply:
As an alternative, it may make sense to flatten `filesByPattern` and deduplicate the result beforehand